consensus/interface.go

```diff
@@ -47,6 +47,22 @@ type SubroundHandler interface {
 	IsInterfaceNil() bool
 }
 
+// SubRoundStartExtraSignatureHandler defines an extra signer during start subround in a consensus process
+type SubRoundStartExtraSignatureHandler interface {
+	Reset(pubKeys []string) error
+	Identifier() string
+	IsInterfaceNil() bool
+}
+
+// SubRoundSignatureExtraSignatureHandler defines an extra signer during signature subround in a consensus process
+type SubRoundSignatureExtraSignatureHandler interface {
+	CreateSignatureShare(header data.HeaderHandler, selfIndex uint16, selfPubKey []byte) ([]byte, error)
```

Comment:
Every new data to be signed should be added in the header. Otherwise, we need to change mostly everything from consensus. Whole code for subrounds is based on sending/veriyfing headers. If we need extra data to be signed in the future, it should be also added in the header.

Reply:
for the new code _Whole code for subrounds is based on sending/veriyfing headers_  
line 187 and 188 create signatures based on []byte and the code before didn't work with headers.

Reply:
And you couldn't directly pass the necessary data to be signed here anyway.
If you have multiple signers, how would you know what data to call them with to be signed? 

This way, each signer selects its relevant data from the header

Reply:
> I was referring the subrounds mechanism

I was refering to the actual `CreateSignatureShare(header data.HeaderHandler, selfIndex uint16, selfPubKey []byte) ([]byte, error)` where my comment was. 


> Check for example "sanity checks" from func (sr *subroundSignature) receivedSignature like IsConsensusDataSet, where the management of consensus data is based on headers.

Is it relevant for a signer? 

My comment was specific for a sign:  

> could the signature be done on a []byte and not on a header

Reply:
> If you have multiple signers, how would you know what data to call them with to be signed?

Maybe it's not the job of the signMethod to know that. It could have another method: GetSigningData that returns the []byte on which the actual signature has to be made